src/rcc.rs

```diff
@@ -318,6 +330,59 @@ impl CFGR {
     }
 }
 
+pub struct LSE {
+    _0: ()
+}
+
+impl LSE {
+    /// Start the LSE clock with a frequency of 32.768 kHz
+    pub fn freeze(self, _bkp: &BackupDomain) -> Lse {
+        // NOTE: Safe because only the BDCR is written to and it is not
+        // used anywhere else.
+        let rcc = unsafe { &*RCC::ptr() };
+
+        rcc.bdcr.modify(|_, w| {
+            w
+                // Enable external low speed oscilator
+                .lseon().set_bit()
+                // Enable the RTC
+                .rtcen().set_bit()
+                // Set the source of the RTC to LSE
+                .rtcsel().lse()
+        });
+
+        // NOTE: Chosing another frequency requires an external oscilator
+        Lse{freq: Hertz(32768)}
```

Comment:
No, starting LSE need an external oscilator, LSE is for LowSpeed External (oscilator). It's for example present on the blue pill (the  black rectangular thing) but not on the black pill. Having something different from a 32k oscilator must be anecdotal.

Reply:
I thought so too at first, but this is what the documentation says in section 8.2.4:

> The LSE crystal is a 32.768 kHz Low Speed External crystal or ceramic resonator. It has the advantage providing a low-power but highly accurate clock source to the real-time clock peripheral (RTC) for clock/calendar or other timing functions.
> 
> The LSE crystal is switched on and off using the LSEON bit in Backup domain control register (RCC_BDCR)